source/mir/math/sum.d

```diff
@@ -1667,39 +1642,91 @@ Returns:
     The sum of all the elements in the range r.
 +/
 template sum(F, Summation summation = Summation.appropriate)
```

Comment:
`sum` doesn't work for two-dimensional dynamic arrays, which means that `mean` doesn't either. It might be useful add in a static assert (perhaps in `Summator`) that provides a better error message. 
```
auto x = [[1.0, 2.0], [3.0, 4.0]];
assert(x.sum == 10);
```

Reply:
added to sumType